From 7e65f1696888a08cdf559f732b9ad1cf1aaba7d5 Mon Sep 17 00:00:00 2001
From: Jonathon Hall <jhall@londontrustmedia.com>
Date: Fri, 20 Dec 2019 09:18:01 -0500
Subject: [PATCH 1/2] Fix termination of long options array

---
 src/daemon.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/daemon.c b/src/daemon.c
index a04e17c..ccfcab5 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -175,7 +175,8 @@ parse_arg(int argc, char **argv, hsk_options_t *opt) {
 #ifndef _WIN32
     { "daemon", no_argument, NULL, 'd' },
 #endif
-    { "help", no_argument, NULL, 'h' }
+    { "help", no_argument, NULL, 'h' },
+    { NULL, 0, 0, 0 }
   };
 
   int longopt_idx = -1;
-- 
2.7.4

