From 3ec7e1cd3eaeef688f39850801dcf5b012f5dc19 Mon Sep 17 00:00:00 2001
From: Jonathon Hall <jhall@londontrustmedia.com>
Date: Thu, 9 May 2019 12:32:31 -0400
Subject: [PATCH 08/14] Fix declaration of after_read (was incorrect on LLP64)

---
 src/pool.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pool.c b/src/pool.c
index f52286f..8956d49 100644
--- a/src/pool.c
+++ b/src/pool.c
@@ -108,7 +108,7 @@ static void
 after_write(uv_write_t *req, int status);
 
 static void
-after_read(uv_stream_t *stream, long int nread, const uv_buf_t *buf);
+after_read(uv_stream_t *stream, ssize_t nread, const uv_buf_t *buf);
 
 static void
 after_close(uv_handle_t *handle);
@@ -1719,7 +1719,7 @@ alloc_buffer(uv_handle_t *handle, size_t size, uv_buf_t *buf) {
 }
 
 static void
-after_read(uv_stream_t *stream, long int nread, const uv_buf_t *buf) {
+after_read(uv_stream_t *stream, ssize_t nread, const uv_buf_t *buf) {
   hsk_peer_t *peer = (hsk_peer_t *)stream->data;
 
   if (!peer)
-- 
2.7.4

