From 72999bbb472a91bd71d19b214c874c2b3d03c7d7 Mon Sep 17 00:00:00 2001
From: Jonathon Hall <jhall@londontrustmedia.com>
Date: Thu, 9 May 2019 16:21:16 -0400
Subject: [PATCH 10/14] Fix pool shutdown

---
 src/pool.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/pool.c b/src/pool.c
index 8956d49..29411ff 100644
--- a/src/pool.c
+++ b/src/pool.c
@@ -333,8 +333,6 @@ hsk_pool_close(hsk_pool_t *pool) {
   if (uv_timer_stop(&pool->timer) != 0)
     return HSK_EFAILURE;
 
-  hsk_pool_uninit(pool);
-
   return HSK_SUCCESS;
 }
 
@@ -343,6 +341,12 @@ hsk_pool_destroy(hsk_pool_t *pool) {
   if (!pool)
     return HSK_EBADARGS;
 
+  int rc = hsk_pool_close(pool);
+  if (rc != HSK_SUCCESS) {
+    hsk_pool_log(pool, "failed to close pool: %s\n", hsk_strerror(rc));
+    return rc;
+  }
+
   hsk_pool_free(pool);
 
   return HSK_SUCCESS;
-- 
2.7.4

